pkg/util/converter/prom.go

```diff
@@ -0,0 +1,404 @@
+					}
+					f.Append(fmt.Sprintf("%v", v))
+					if f.Len() > max {
+						max = f.Len()
+					}
+				}
+
+				for _, f := range lookup {
+					if f.Len() != max {
+						f.Append("") // no matching label
+					}
+				}
+			}
+
+		default:
+			{
+				ext := iter.ReadAny()
+				v := fmt.Sprintf("%v", ext)
+				stringField.Append(v)
+			}
+		}
+	}
+
+	if rsp.Frames == nil {
+		rsp.Frames = data.Frames{data.NewFrame("", stringField)}
+	}
+
+	return rsp
+}
+
+// For consistent ordering read values to an aray not a map
+func readLabelsAsPairs(iter *jsoniter.Iterator) [][2]string {
+	pairs := make([][2]string, 0, 10)
+	for k := iter.ReadObject(); k != ""; k = iter.ReadObject() {
+		pairs = append(pairs, [2]string{k, iter.ReadString()})
+	}
+	return pairs
+}
+
+func readLabelsOrExemplars(iter *jsoniter.Iterator) (*data.Frame, [][2]string) {
+	pairs := make([][2]string, 0, 10)
+	labels := data.Labels{}
+	var frame *data.Frame
+
+	for l1Field := iter.ReadObject(); l1Field != ""; l1Field = iter.ReadObject() {
+		switch l1Field {
+		case "seriesLabels":
+			iter.ReadVal(&labels)
+		case "exemplars":
+			lookup := make(map[string]*data.Field)
+			timeField := data.NewFieldFromFieldType(data.FieldTypeTime, 0)
+			valueField := data.NewFieldFromFieldType(data.FieldTypeFloat64, 0)
+			valueField.Name = labels["__name__"]
+			delete(labels, "__name__")
+			valueField.Labels = labels
+			frame = data.NewFrame("", timeField, valueField)
+			for iter.ReadArray() {
+				for l2Field := iter.ReadObject(); l2Field != ""; l2Field = iter.ReadObject() {
+					switch l2Field {
+					// nolint:goconst
+					case "value":
+						v, _ := strconv.ParseFloat(iter.ReadString(), 64)
+						valueField.Append(v)
+
+					case "timestamp":
+						ts := timeFromFloat(iter.ReadFloat64())
+						timeField.Append(ts)
+
+					case "labels":
+						max := 0
+						for _, pair := range readLabelsAsPairs(iter) {
+							k := pair[0]
+							v := pair[1]
+							f, ok := lookup[k]
+							if !ok {
+								f = data.NewFieldFromFieldType(data.FieldTypeString, 0)
+								f.Name = k
+								lookup[k] = f
+								frame.Fields = append(frame.Fields, f)
+							}
+							f.Append(v)
+							if f.Len() > max {
+								max = f.Len()
+							}
+						}
+						for _, f := range lookup {
+							if f.Len() != max {
+								f.Append("") // no matching label
+							}
+						}
+
+					default:
+						iter.Skip()
+						frame.AppendNotices(data.Notice{
+							Severity: data.NoticeSeverityError,
+							Text:     fmt.Sprintf("unable to parse key: %s in response body", l2Field),
+						})
+					}
+				}
+			}
+		default:
+			v := fmt.Sprintf("%v", iter.Read())
+			pairs = append(pairs, [2]string{l1Field, v})
+		}
+	}
+
+	return frame, pairs
+}
+
+func readMatrixOrVector(iter *jsoniter.Iterator) *backend.DataResponse {
+	rsp := &backend.DataResponse{}
+
+	for iter.ReadArray() {
+		timeField := data.NewFieldFromFieldType(data.FieldTypeTime, 0) // for now!
```

Comment:
for streams-data the timefield has a name, could we name this too? either "Time" or "time", i don't know which is "more standard".

Reply:
👍 set it to "Time" (using the golang constant)